iceoryx_posh/source/runtime/posh_runtime_impl.cpp

```diff
@@ -185,12 +185,22 @@ PoshRuntimeImpl::requestPublisherFromRoudi(const IpcMessage& sendBuffer) noexcep
         std::string IpcMessage = receiveBuffer.getElementAtIndex(0U);
 
         if (stringToIpcMessageType(IpcMessage.c_str()) == IpcMessageType::CREATE_PUBLISHER_ACK)
-
         {
             segment_id_underlying_t segmentId{0U};
-            convert::fromString(receiveBuffer.getElementAtIndex(2U).c_str(), segmentId);
             UntypedRelativePointer::offset_t offset{0U};
-            convert::fromString(receiveBuffer.getElementAtIndex(1U).c_str(), offset);
+            auto segment_id_result =
+                convert::from_string<segment_id_underlying_t>(receiveBuffer.getElementAtIndex(2U).c_str());
+            auto offset_result =
+                convert::from_string<UntypedRelativePointer::offset_t>(receiveBuffer.getElementAtIndex(1U).c_str());
+
+            if (!segment_id_result.has_value() || !offset_result.has_value())
+            {
+                IOX_LOG(ERROR, "segment_id and/or offset conversion failed");
+                return err(IpcMessageErrorType::SEGMENT_ID_OR_OFFSET_CONVERSION_FAILURE);
+            }
+
+            segmentId = segment_id_result.value();
+            offset = offset_result.value();
             auto ptr = UntypedRelativePointer::getPtr(segment_id_t{segmentId}, offset);
```

Comment:
This looks like it could be refactored to a separate method and used in all the places below to reduce code duplication.

Reply:
Sure.

Reply:
See [dd422c2](https://github.com/eclipse-iceoryx/iceoryx/pull/2135/commits/dd422c220f13fcf4f0c445068738938b8a4be6c9).